bootstrap-extensions/src/main/java/de/agilecoders/wicket/extensions/markup/html/bootstrap/form/tempusdominus/TempusDominusConfig.java

```diff
@@ -0,0 +1,285 @@
+
+import static java.time.format.DateTimeFormatter.ISO_INSTANT;
+import static java.time.format.DateTimeFormatter.ISO_LOCAL_DATE;
+import static java.time.format.DateTimeFormatter.ISO_LOCAL_DATE_TIME;
+import static java.time.format.DateTimeFormatter.ISO_LOCAL_TIME;
+import static java.time.format.DateTimeFormatter.ISO_ZONED_DATE_TIME;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.ZonedDateTime;
+import java.util.Date;
+import java.util.Locale;
+import java.util.Map;
+import java.util.function.Consumer;
+
+import de.agilecoders.wicket.jquery.AbstractConfig;
+import de.agilecoders.wicket.jquery.IKey;
+import de.agilecoders.wicket.jquery.util.Json.RawValue;
+
+/**
+ * Config of datetime picker plugin.
+ *
+ * @see <a href="https://getdatepicker.com/6/options/">JS widget options</a>
+ */
+public class TempusDominusConfig extends AbstractConfig {
+    private static final long serialVersionUID = 1L;
+
+    private static final IKey<Boolean> AllowInputToggle = newKey("allowInputToggle", false);
+    private static final IKey<RawValue> Container = newKey("container", null);
+    private static final IKey<Boolean> DateRange = newKey("dateRange", false);
+    private static final IKey<RawValue> DefaultDate = newKey("defaultDate", null);
+    private static final IKey<TempusDominusDisplayConfig> Display = newKey("display", null);
+    private static final IKey<Boolean> KeepInvalid = newKey("keepInvalid", false);
+    private static final IKey<Map<String, Object>> Meta = newKey("meta", null);
+    private static final IKey<Boolean> MultipleDates = newKey("multipleDates", false);
+    private static final IKey<String> MultipleDatesSeparator = newKey("multipleDatesSeparator", null);
+    private static final IKey<Boolean> PromptTimeOnDateChange = newKey("promptTimeOnDateChange", false);
+    private static final IKey<Integer> PromptTimeOnDateChangeTransitionDelay = newKey("promptTimeOnDateChangeTransitionDelay", null);
+    private static final IKey<TempusDominusRestrictionsConfig> Restrictions = newKey("restrictions", null);
+    private static final IKey<Integer> Stepping = newKey("stepping", null);
+    private static final IKey<Boolean> UseCurrent = newKey("useCurrent", true);
+    private static final IKey<RawValue> ViewDate = newKey("viewDate", null);
+
+    private TempusDominusLocalizationConfig localization = new TempusDominusLocalizationConfig();
+
+    /**
+     * Construct config
+     */
+    public TempusDominusConfig() {
+        put(Display, new TempusDominusDisplayConfig());
+        put(Restrictions, new TempusDominusRestrictionsConfig());
+    }
+
+    /**
+     * Will add restrictions based on class LocalDate/LocalTime/LocalDateTime etc.
+     *
+     * @param clazz Class of DateTime object to set restrictions
+     * @return current instance
+     */
+    public <T> TempusDominusConfig withClass(Class<T> clazz) {
```

Comment:
Should `T` extend `Temporal` ? Otherwise the user could pass anything.

Reply:
Unfortunately this can be `Temporal` OR `java.util.Date` I'm not sure how bounds can be set here :(

Reply:
Do you have a use case in your app for using j.u.Date ? If NOT then I'd recommend to support only the new Datetime APIs

Reply:
Hopefully now everything is better :)